### PR TITLE
fix(crc32c): escape the version dot and anchor the pyenv check regex

### PR DESCRIPTION
**Pull request by chalmerlowe**
### Problem
The OS X python wheel build script (`build_python_wheel.sh`) checks if a Python version is installed in `pyenv` before building:

```bash
if [ -z "$(pyenv versions --bare | grep $version)" ]
```

When $version is 3.14, this expands to grep 3.14. Because the dot (.) is not escaped, it is treated as a wildcard and will also match the substring 3.13.14 (which contains 3.<any char>14 at the end). This results in a false positive, causing the script to skip compiling Python 3.14. The script then fails when executing pyenv shell 3.14 with a "version not installed" error.

### Solution
Escape the version dots in bash: `${version//./\.}.`
Anchor the grep match with line start (`^`) and word boundaries (`\b`) to ensure it only matches the exact version line (e.g. `^3\.14\b` instead of matching `3.13.14`).
Applied this fix to both `build_python_wheel.sh` and `publish_python_wheel.sh`.

Blocks: https://github.com/googleapis/google-cloud-python/pull/17642

### CODE REVIEW

File: packages/google-crc32c/scripts/osx/build_python_wheel.sh [hunk 1]
```diff
@@ -39,8 +39,9 @@ eval "$(pyenv init --path)"
 
 install_python_pyenv() {
     version=$1
+    escaped_version="${version//./\.}"
 
-    if [ -z "$(pyenv versions --bare | grep $version)" ]; then
+    if [ -z "$(pyenv versions --bare | grep "^${escaped_version}\b")" ]; then
```
[the remark this answers is not in the document]
Comment (chalmerlowe) [reply packages/google-crc32c/scripts/osx/build_python_wheel.sh]: resolved.

File: packages/google-crc32c/scripts/osx/publish_python_wheel.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 set -eo pipefail
 
-if [ -z "$(pyenv versions --bare | grep 3.10)" ]; then
+if [ -z "$(pyenv versions --bare | grep "^3\.10\b")" ]; then
```
[the remark this answers is not in the document]
Comment (chalmerlowe) [reply packages/google-crc32c/scripts/osx/publish_python_wheel.sh]: Resolved.

File: packages/google-crc32c/scripts/osx/build_python_wheel.sh [hunk 3]
```diff
@@ -39,8 +39,8 @@ eval "$(pyenv init --path)"
 
 install_python_pyenv() {
     version=$1
-
-    if [ -z "$(pyenv versions --bare | grep $version)" ]; then
+    escaped_version="${version//./\.}"
```
Comment (chalmerlowe) [inline packages/google-crc32c/scripts/osx/build_python_wheel.sh:43]:
```suggestion
    # escapes the dot in the version number to avoid regex expansion issues.
    escaped_version="${version//./\.}"
```

**Comment by chalmerlowe** [comment]
Re: the failing cover test:

When `google-crc32c` gets a change to core code, it triggers as a diff in CI/CD and attempts to run `unittests` in the `unit` job in [`workflows/unittest.yml`](https://github.com/googleapis/google-cloud-python/blob/main/.github/workflows/unittest.yml).
Within that job the `unit` nox session is run via a call to `ci/run_conditional_tests.sh`.

However... in the crc32c noxfile.py it uses `session.skip` for unit tests because they are [not implemented](https://github.com/googleapis/google-cloud-python/blob/3619b29e10ae04623d101808cb98be5edbb483b4/packages/google-crc32c/noxfile.py#L172).

This means **NO coverage results** are produced.
Which means when the `unittest.yml` file rolls down to the step that runs `coverage`, it blows up.


> [!NOTE]
> running coverage is entirely managed in the `unittest.yml` file.
It does NOT run as a nox session defined in the crc32c noxfile.py. In fact there is no `cover` session, such as we have in other packages.
Thus we can't just go in and tell nox to skip the `cover` session the way we did with the `unit` session. 

OPTIONS under consideration:
1. Add a filter in the `unittest.yml` to ignore crc32c when trying launch `coverage`. We don't yet know how that works in terms of publishing a "success" result so that the CI/CD check succeeds.
2. Add a dummy test (with dummy output) in the crc32c noxfile unit test to create coverage files that reflect 100% coverage for crc32c.

**Comment by chalmerlowe** [comment]
> Re: the failing cover test:
> 
An alternate approach was taken and merged into `main` that solves the majority of the issues but still had a minor problem with `google-crc32c`. this PR addresses that last edge case.

**Review by parthea — APPROVED** [review]
